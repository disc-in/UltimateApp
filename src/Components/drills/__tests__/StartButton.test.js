import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import StartButton from '../StartButton';

describe('<StartButton />', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<StartButton text="OK" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls delegates onPress prop', async () => {
    const onPress = jest.fn();

    const { getByTestId } = render(<StartButton text="Easy" onPress={onPress} />);

    await fireEvent.press(getByTestId('startButton'));

    expect(onPress).toHaveBeenCalled();
  });
});
