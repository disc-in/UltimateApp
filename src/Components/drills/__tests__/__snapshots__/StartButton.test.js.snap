// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StartButton /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2a9e91",
      "borderRadius": 50,
      "height": 100,
      "justifyContent": "center",
      "margin": 20,
      "opacity": 1,
      "width": 100,
    }
  }
  testID="startButton"
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    OK
  </Text>
</View>
`;
